vhost-device-scmi/src/devices/iio.rs

```diff
@@ -286,6 +296,97 @@ impl SensorT for IIOSensor {
         }
         Ok(result)
     }
+
+    fn get_notify_fd(&self) -> Option<RawFd> {
+        self.sensor.notify_dev.as_ref().map(|fd| fd.as_raw_fd())
+    }
+
+    fn reading_device(&self, device_index: u32) -> DeviceResult {
+        let mut result = vec![];
+
+        //Here we should create buffer which length is corresponding to iio:device;
+        //future change to self.get_data_format
+        //cat /sys/bus/iio/devices/iio:deviceX/scan_elements/in_XXX_type.
+        //for example le:s16/16>>0 , [x,y,z,(t)] each is signed 16, and Little Endian
+        //so one sample length is (3 or 4)*(16/8) byte
+        let sample_buffer_len = 8;
+        let mut buffer = vec![0u8; sample_buffer_len];
```

Comment:
This feels quite unsafe. How about checking that the value type is indeed le:s16 and returning an error in case it is not, rather than outputting nonsense in such a case?

Reply:
Is it better to read from "in_XXX_type" firstly? And re-define related arguments according to result?
I will try to refactor it.

Reply:
Yes, that would be best.